libpromises/conversion.c

```diff
@@ -211,11 +211,11 @@ char *Rlist2String(Rlist *list, char *sep)
 
     for (rp = list; rp != NULL; rp = rp->next)
     {
-        strcat(line, RlistScalarValue(rp));
+        strlcat(line, RlistScalarValue(rp), sizeof(line));
 
         if (rp->next)
         {
-            strcat(line, sep);
+            strlcat(line, sep, sizeof(line));
```

Comment:
I'm not cherry-picking this one because this is a very generic utility function and who knows what truncation can cause. Please add a `Log(ERR, "internal limit reached ...")` or similar in the case truncation happens.